### PR TITLE
KAFKA-12449: Remove deprecated WindowStore#put

### PR DESCRIPTION
**Pull request by jeqo**
Completes KIP-474 https://cwiki.apache.org/confluence/x/kcviBg by removing deprecated WindowStore#put(k, v). 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java [hunk 1]
```diff
@@ -710,73 +710,9 @@ public void testPutAndFetchAfter() {
         assertNull(entriesByKey.get(6));
     }
 
-    @Test
-    @SuppressWarnings("deprecation")
-    public void testPutSameKeyTimestamp() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java:715]: Why removing the whole test?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java [hunk 2]
```diff
@@ -710,73 +710,9 @@ public void testPutAndFetchAfter() {
...
-
-        final Map<Integer, Set<String>> entriesByKey = entriesByKey(changeLog, startTime);
-
-        assertEquals(Utils.mkSet("zero@0", "zero@0", "zero+@0", "zero++@0"), entriesByKey.get(0));
-    }
-
     @Test
     public void shouldCloseOpenIteratorsWhenStoreIsClosedAndNotThrowInvalidStateStoreExceptionOnHasNext() {
-        setCurrentTime(0);
+        context.setRecordContext(createRecordContext(0));
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java]: Do we need to set the record context?

**Review by mjsax — COMMENTED** [review]
Overall LGTM. Thanks for the PR.

**Comment by mjsax** [comment]
Seem this PR is related to KIP-474 https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=115526545.

We should rather tag the KIPs that did the deprecation of methods (on KIP-667 that is unnecessary)

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java (hunk 1 above)
Comment (jeqo) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java:715]: My mistake, I thought it was repeated. Will add it back.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java (hunk 2 above)
Comment (jeqo) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java]: Seems that we don't :) will remove this line.

**Comment by jeqo** [comment]
~~Totally missed that this was proposed and accepted in KIP-474. Hope is not stepping into an existing implementation.~~

Just realized the KIP is just about deprecating this method, and this is just completing it with the removal 😅. Turned the sub-task into a task in JIRA and linked it to the right KIP.

**Review by mjsax — APPROVED** [review]
Overall LGTM.

Can you add a bullet point to `docs/streams/upgrade-guide.html` for this PR (we list all APIs that we remove there).

Also PR needs a rebase.

**Comment by mjsax** [comment]
Thanks @jeqo! Merged to `trunk`.

**Pull request merged**